examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,298 @@
+## Evaluation Utilities
+
+Image Super-Resolution or Denoiser models can be evaluated with PSNR, SSIM metrics for
+perceptual quality assessments but our task of **Text Restoration** requires a bit more,
+our objective is to Enhance simple OCR accuracy engine where our model can act as a
+**Pre-Processor**, to serve the purpose we will use <a
+href="https://towardsdatascience.com/evaluating-ocr-output-quality-with-character-error-ra
+te-cer-and-word-error-rate-wer-853175297510">WER(Word Error Rate) </a> as a metric to
+evaluate the real-world performance. The following function ```generate_results```  will
+help plot the visual comparisons of outputs given the **Raw Noisy Input** and **Model
+Restored Input** to tesseract OCR engine.
+"""
+
+
+def word_error_rate(noise_image, reference_image, prediction_image):
+    reference = pytesseract.image_to_string(reference_image)
+    noiseimage_ocr_out = pytesseract.image_to_string(noise_image)
+    predimage_ocr_out = pytesseract.image_to_string(prediction_image)
+    noise_WER_score = fastwer.score_sent(noiseimage_ocr_out, reference)
+    pred_WER_score = fastwer.score_sent(predimage_ocr_out, reference)
+    return noise_WER_score, pred_WER_score
+
+
+def generate_results(file):
+    clean_gt_file = file.replace("source", "target")
+    source_image = preprocess(file)
+    noise_ground_truth = keras.utils.array_to_img(denormalize(source_image))
+    clean_ground_truth = keras.utils.array_to_img(
+        denormalize(preprocess(clean_gt_file))
+    )
+
+    source_image = tf.expand_dims(source_image, axis=0)
+    prediction = denoiser_net.predict(source_image)
+    prediction = keras.utils.array_to_img(denormalize(prediction[0]))
+
+    noise_error_rate, reconstructed_error_rate = word_error_rate(
+        noise_image=noise_ground_truth,
+        reference_image=clean_ground_truth,
+        prediction_image=prediction,
+    )
+
+    fig, axes = plt.subplots(1, 2, figsize=(18, 6))
+    axes[0].imshow(noise_ground_truth)
+    axes[0].set_xlabel(f"Input-WER score : {round(noise_error_rate,2)}")
+    axes[1].imshow(prediction)
+    axes[1].set_xlabel(f"Reconstructed-WER score : {round(reconstructed_error_rate,2)}")
+    #     axes[2].imshow(clean_ground_truth); axes[2].set_xlabel("clean_ground-truth")
+    fig.suptitle(
+        "Visual comparison and WER(Word Error Rate) Evaluation charts",
+        weight="bold",
+        fontsize=12,
+    )
+    return noise_ground_truth, prediction, clean_ground_truth
+
+
+"""
+## Visualizing Results 
+"""
+
+test_images = [np.random.choice(val_source) for i in range(6)]
+for f in test_images:
+    generate_results(f)
+
+"""
+## Conclusion
+
+We are able to observe using the model as a Pre-Processor enables significant amount of
+improvement in Word Error Rate and perceptual quality of the image, the experiment can be
+extended to different Image-to-Image applications, also Training with large datasets and
+more epochs can elevate the performance of the model further.
```

Comment:
@anish9 same here. For example, the first sentence is too long to read. Break it down, make it simple and use full stop when necessary.